scim2-sdk-common/src/main/java/com/unboundid/scim2/common/messages/PatchOperation.java

```diff
@@ -126,6 +127,39 @@ public abstract class PatchOperation
   @Nullable
   private final Path path;
 
+  /**
+   * This represents a property that, when enabled, will always append new
+   * values when processing ADD operations with a value filter. This is used for
+   * multi-valued attributes such as {@code emails}. For example, for the
+   * following patch request:
+   * <pre>
+   *   {
+   *     "schemas": [ "urn:ietf:params:scim:api:messages:2.0:PatchOp" ],
+   *     "Operations": [
+   *       {
+   *         "op": "add",
+   *         "path": "emails[type eq \"work\"].display",
+   *         "value": "newEmail@example.com"
+   *       }
+   *     ]
+   *   }
+   * </pre>
+   *
+   * When this property is enabled and the above patch request is applied, the
+   * following JSON will be appended to the {@code emails} of the user resource:
+   * <pre>
+   *   {
+   *     "type": "work",
+   *     "display": "newEmail@example.com"
+   *   }
+   * </pre>
+   *
+   * If this property is disabled, then the {@code display} field will be added
+   * to the resource's existing work email, if it exists. If the work email does
+   * not exist, then a new value will be appended.
+   */
+  public static boolean APPEND_NEW_PATCH_VALUES_PROPERTY = true;
```

Comment:
I'm uncertain about making this a public static field on the class, since it doesn't seem like they're used for configuration like this elsewhere in this library. Also, I personally would expect this kind of logic-changing behavior to be configurable on an operation-by-operation basis, which doesn't seem possible when using a public static field.

I _can_ see why we would want to use a public static field, because the cleanest option in my opinion -- adding a new field to the AddOperation class that the `applyAddWithValueFilter` method checks for -- means we're no longer sticking to the SCIM RFC as closely as we were. But if it's possible to do that, I think that might be worth considering.

If we don't want to do that, I would recommend making this field belong to the `AddOperation` class instead of the `PatchOperation` class, since the `RemoveOperation` and `ReplaceOperation` classes don't need it.